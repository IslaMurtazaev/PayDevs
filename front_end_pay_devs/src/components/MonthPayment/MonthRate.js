import React from "react";
import { Link } from "react-router-dom";
import { history } from "../../index";

const MonthRate = props => {
  return (
    <div>
      <Link
        to={`${history.location.pathname}/Monthly/${props.id}/workedDays`}
      >
        <h3>{props.rate}</h3>
      </Link>
      <button onClick={() => props.onRemove(props.id)}>Remove rate</button>
    </div>
  );
};

export default MonthRate;
