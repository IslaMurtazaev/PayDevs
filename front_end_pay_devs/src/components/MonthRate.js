import React from "react";

const MonthRate = props => {
  return (
    <div>
      <h3>{props.rate}</h3>
      <button onClick={() => props.onRemove(props.id)}>Remove rate</button>
    </div>
  );
};

export default MonthRate;
