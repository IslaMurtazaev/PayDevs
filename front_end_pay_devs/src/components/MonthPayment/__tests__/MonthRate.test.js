import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import MonthRate from "../MonthlyRate";

describe("<MonthRate />", () => {
  const monthPayment = { rate: 150, id: 1 };

  it("renders rate based on props", () => {
    const monthRate = shallow(<MonthRate monthPayment={monthPayment} />);
    expect(monthRate.find(".rate").text()).toBe("150/per day");
  });

  it("renders remove button based on props", () => {
    const onRemoveSpy = jest.fn();
    const monthRate = shallow(<MonthRate monthPayment={monthPayment} onRemove={onRemoveSpy} />);
    monthRate.find(".removeRate").simulate("click");
    expect(onRemoveSpy).toBeCalledWith(1);
  });

  it("matches previous snapshot", () => {
    expect(toJson(shallow(<MonthRate monthPayment={monthPayment} />))).toMatchSnapshot();
  })
});
