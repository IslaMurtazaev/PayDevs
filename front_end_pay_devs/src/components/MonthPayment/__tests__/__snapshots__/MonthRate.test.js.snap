// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthRate /> matches previous snapshot 1`] = `
<div>
  <Link
    replace={false}
    to="/project/undefined/Monthly/1/workedDay"
  >
    <span
      className="rate"
    >
      150
      /per day
    </span>
  </Link>
  <button
    className="removeRate"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
